### ISSUE TITLE
Why do ModernBERT embeddings show similar cosine similarities (>0.90) for unrelated sentences?

### ISSUE THREAD

**Issue by kdbeall** [AnswerDotAI/ModernBERT#214]
Here's my code to reproduce.
I get a cosine similarity of 0.93 between
"Quantum physics explores the behavior of matter at the subatomic level" and
"The new recipe for chocolate chip cookies requires brown sugar and vanilla extract"

Thanks!

```python
import torch
import numpy as np
from transformers import AutoModel, AutoTokenizer

# Load model and tokenizer
tokenizer = AutoTokenizer.from_pretrained("answerdotai/ModernBERT-base")
model = AutoModel.from_pretrained("answerdotai/ModernBERT-base")

# Two sentences with vastly different topics
text1 = "Quantum physics explores the behavior of matter at the subatomic level."
text2 = "The new recipe for chocolate chip cookies requires brown sugar and vanilla extract."

# Tokenize both texts
inputs1 = tokenizer(text1, return_tensors="pt", padding=True, truncation=True)
inputs2 = tokenizer(text2, return_tensors="pt", padding=True, truncation=True)

# Get model outputs
with torch.no_grad():
    outputs1 = model(**inputs1)
    outputs2 = model(**inputs2)

# Get CLS token embeddings from last layer
embedding1 = outputs1.last_hidden_state[:, 0, :].squeeze().numpy()
embedding2 = outputs2.last_hidden_state[:, 0, :].squeeze().numpy()

# Normalize the embeddings
normalized_embedding1 = embedding1 / np.linalg.norm(embedding1)
normalized_embedding2 = embedding2 / np.linalg.norm(embedding2)

# Calculate cosine similarity
similarity = np.dot(normalized_embedding1, normalized_embedding2)

print(f"Embedding 1 shape: {embedding1.shape}")
print(f"Embedding 2 shape: {embedding2.shape}")
print(f"Cosine similarity between the two sentences: {similarity}")
```

**Comment by kdbeall** [comment]
I think I have to train my own model based on
https://github.com/AnswerDotAI/ModernBERT/blob/main/examples/train_st.py

But yeah, this should be better documented.

**Comment by jihobak** [comment]
As you said, you need additional training to get a good result because this model comes only from MLM training. In my opinion, we don’t need to write it in a document; it’s natural.

**Comment by NohTow** [comment]
>I think I have to train my own model based on
https://github.com/AnswerDotAI/ModernBERT/blob/main/examples/train_st.py

Yes, indeed, or you can use models people have already trained for you as [ModernBERT-embed-base](https://huggingface.co/nomic-ai/modernbert-embed-base) (or its [large variant](https://huggingface.co/lightonai/modernbert-embed-large)) or [the GTE version of ModernBERT](https://huggingface.co/Alibaba-NLP/gte-modernbert-base).

> But yeah, this should be better documented.

I mean, it is written in the ["Usage" section of the readme of the model](https://huggingface.co/answerdotai/ModernBERT-base#usage) quite explicitely:
> Since ModernBERT is a Masked Language Model (MLM), you can use the fill-mask pipeline or load it via AutoModelForMaskedLM. To use ModernBERT for downstream tasks like classification, retrieval, or QA, fine-tune it following standard BERT fine-tuning recipes.

Not sure what we can do better.

**Comment by kdbeall** [comment]
@NohTow Thanks! An exact stand alone paragraph could be a good idea. Some of us are ADHD you know :)
Apologies for nitpicking here.
